cpp/src/arrow/type.h

```diff
@@ -1546,6 +1581,97 @@ std::shared_ptr<Schema> schema(
 
 /// @}
 
+/// \brief Convenience class to incrementally construct/merge schemas.
+///
+/// This class amortizes the cost of validating field name conflicts by
+/// maintaining the mapping. The caller also controls the conflict resolution
+/// scheme.
+class SchemaBuilder {
+ public:
+  // Indicate how field conflict(s) should be resolved when building a schema. A
+  // conflict arise when a field is added to the builder and one or more field(s)
+  // with the same name already exists.
+  enum ConflictPolicy {
+    // Ignore the conflict and append the field. This is the default behavior of the
+    // Schema constructor and the `arrow::schema` factory function.
+    APPEND = 0,
+    // Replace the existing field with the newer one.
+    REPLACE,
```

Comment:
I'm tempted to change this with `KEEP_OURS`, `KEEP_THEIRS` like in git merge.

Reply:
I prefer `REPLACE` and `IGNORE`

Reply:
Settled to `IGNORE` and `REPLACE`.